lib/abstract_format.ml

```diff
@@ -909,6 +915,43 @@ and guard_test_of_sf sf : (guard_test_t, err_t) Result.t =
      let%bind rhs = sf_rhs |> guard_test_of_sf |> track ~loc:[%here] in
      GuardTestBinOp {line; op; lhs; rhs} |> return
 
+  (* a record creation : #user{name = "Taro", admin = true} *)
+  | Sf.Tuple (4, [Sf.Atom "record";
+                  Sf.Integer line;
+                  Sf.Atom name;
+                  Sf.List sf_record_fields]) ->
+     let field_of_sf sf =
+       begin match sf with
+       | Sf.Tuple (4, [Sf.Atom "record_field";
+                       Sf.Integer line;
+                       sf_atom_or_wildcard;
+                       sf_guard_test]) ->
+          let%bind field_name = atom_or_wildcard_of_sf sf_atom_or_wildcard in
+          let%bind rhs = guard_test_of_sf sf_guard_test in
+          (line, field_name, rhs) |> return
+       | _ ->
+          Err.create ~loc:[%here] (Err.Not_supported_absform ("cannot reach here", sf)) |> Result.fail
+       end
+     in
+     let%bind record_fields = sf_record_fields |> List.map ~f:field_of_sf |> Result.all |> track ~loc:[%here] in
+     GuardTestRecord {line; name; record_fields} |> return
+
+  (* a record field access : U#user.name *)
+  | Sf.Tuple (5, [Sf.Atom "record_field";
+                  Sf.Integer line;
+                  sf_record;
+                  Sf.Atom name;
+                  Sf.Tuple (3, [Sf.Atom "atom"; _; Sf.Atom field_name])]) ->
```

Comment:
Please don't leave line number.

Could you add `line_field_name` field to `GuardTestRecordFieldAccess`?

Reply:
yes